hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/wal/TestAsyncLogRolling.java

```diff
@@ -51,6 +60,64 @@ public static void setUpBeforeClass() throws Exception {
     AbstractTestLogRolling.setUpBeforeClass();
   }
 
+  public static class SlowSyncLogWriter extends AsyncProtobufLogWriter {
+
+    public SlowSyncLogWriter(EventLoopGroup eventLoopGroup, Class<? extends Channel> channelClass) {
+      super(eventLoopGroup, channelClass);
+    }
+
+    @Override
+    public CompletableFuture<Long> sync(boolean forceSync) {
+      CompletableFuture<Long> future = new CompletableFuture<>();
+      super.sync(forceSync).whenCompleteAsync((lengthAfterFlush, error) -> {
+        if (error != null) {
+          future.completeExceptionally(error);
+        } else {
+          try {
+            Thread.sleep(syncLatencyMillis);
```

Comment:
Let's introduce a ScheduledExecutorService and schedule a delayed task for completing the future? Sleeping in a callback is still not a good behavior...

And even if there is an error, we still need to delay the completing?

Reply:
> And even if there is an error, we still need to delay the completing?
Yes, you are right and I have made the changes. Please check if it meets your requirements.